examples/saml-service-provider/src/main/java/example/armeria/server/saml/sp/MyAuthHandler.java

```diff
@@ -0,0 +1,97 @@
+package example.armeria.server.saml.sp;
+
+import static com.linecorp.armeria.server.saml.SamlUtil.getNameId;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+
+import javax.annotation.Nullable;
+
+import org.opensaml.messaging.context.MessageContext;
+import org.opensaml.saml.saml2.core.NameIDType;
+import org.opensaml.saml.saml2.core.Response;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Strings;
+
+import com.linecorp.armeria.common.AggregatedHttpMessage;
+import com.linecorp.armeria.common.HttpData;
+import com.linecorp.armeria.common.HttpHeaderNames;
+import com.linecorp.armeria.common.HttpHeaders;
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.common.MediaType;
+import com.linecorp.armeria.server.ServiceRequestContext;
+import com.linecorp.armeria.server.auth.Authorizer;
+import com.linecorp.armeria.server.saml.SamlNameIdFormat;
+import com.linecorp.armeria.server.saml.SamlSingleSignOnHandler;
+
+import io.netty.handler.codec.http.cookie.Cookie;
+import io.netty.handler.codec.http.cookie.DefaultCookie;
+import io.netty.handler.codec.http.cookie.ServerCookieDecoder;
+import io.netty.handler.codec.http.cookie.ServerCookieEncoder;
+
+final class MyAuthHandler implements Authorizer<HttpRequest>, SamlSingleSignOnHandler {
+    private static final Logger logger = LoggerFactory.getLogger(MyAuthHandler.class);
+
+    /**
+     * Checks whether the request has right permission to access the service. In this example, a {@code Cookie}
+     * header is used to hold the information. If a {@code username} key exists in the {@code Cookie} header,
+     * the request is treated as authenticated.
```

Comment:
Perhaps we should mention that this is purely for a demonstration purpose and a user should perform proper authorization in a real world application?

Reply:
Added it to the class-level comment. What do you think about that?